src/cloud-api-adaptor/test/e2e/common_suite.go

```diff
@@ -615,3 +615,46 @@ func DoTestKbsKeyReleaseForFailure(t *testing.T, e env.Environment, assert Cloud
 
 	NewTestCase(t, e, "DoTestKbsKeyReleaseForFailure", assert, "Kbs key release is failed").WithPod(pod).WithTestCommands(testCommands).Run()
 }
+
+func DoTestRestrictivePolicyBlocksExec(t *testing.T, e env.Environment, assert CloudAssert) {
+	allowAllExceptExecPolicyFilePath := "../../podvm/files/etc/kata-opa/allow-all-except-exec-process.rego"
+	podName := "policy-exec-rejected"
+	pod := NewPodWithPolicy(E2eNamespace, podName, allowAllExceptExecPolicyFilePath)
+
+	testCommands := []TestCommand{
+		{
+			Command:       []string{"ls"},
+			ContainerName: pod.Spec.Containers[0].Name,
+			TestErrorFn: func(err error) bool {
+				if strings.Contains(err.Error(), "failed to exec in container") && strings.Contains(err.Error(), "ExecProcessRequest is blocked by policy") {
+					log.Infof("Exec process was blocked %s", err.Error())
+					return true
+				} else {
+					log.Errorf("Exec process was allowed: %s", err.Error())
+					return false
+				}
+			},
+		},
+	}
+	NewTestCase(t, e, "PodVMwithPolicyBlockingExec", assert, "Pod which blocks Exec Process").WithPod(pod).WithTestCommands(testCommands).Run()
+}
+
+func DoTestPermissivePolicyAllowsExec(t *testing.T, e env.Environment, assert CloudAssert) {
+	allowAllPolicyFilePath := "../../podvm/files/etc/kata-opa/allow-all.rego"
+	podName := "policy-all-allowed"
+	pod := NewPodWithPolicy(E2eNamespace, podName, allowAllPolicyFilePath)
+
+	// Just check there are no errors and something returned
```

Comment:
I got a little bit disconnected of this project and I didn't follow the great changes that you guys made to the e2e test framework. It's amazing to see how easy it got to build new test cases. Thanks!

Reply:
Yeah definitely - this task was way easier than I thought it would be 😃